library/common/extensions/filters/http/platform_extension/bridging_filter.cc

```diff
@@ -0,0 +1,102 @@
+#include "library/common/extensions/filters/http/platform_extension/bridging_filter.h"
+
+#include "envoy/server/filter_config.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+
+#include "library/common/api/external.h"
+#include "library/common/buffer/utility.h"
+#include "library/common/http/header_utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace PlatformExtension {
+
+Http::FilterHeadersStatus mapStatus(envoy_filter_headers_status_t status) {
+  switch (status) {
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE:
+    return Http::FilterHeadersStatus::Continue;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ITERATION:
+    return Http::FilterHeadersStatus::StopIteration;
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE_AND_END_STREAM:
+    return Http::FilterHeadersStatus::ContinueAndEndStream;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ALL_ITERATION_AND_BUFFER:
+    return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
+  default:
+    ASSERT(false, "unrecognized filter status from platform: {}");
+    return Http::FilterHeadersStatus::Continue;
+  }
+}
+
+BridgingFilterConfig::BridgingFilterConfig(
+    const envoymobile::extensions::filters::http::platform_extension::Bridging& proto_config)
+    : name_(proto_config.name()) {}
+
+BridgingFilter::BridgingFilter(BridgingFilterConfigSharedPtr config)
+    : platform_filter_(
+          static_cast<envoy_http_filter*>(Api::External::retrieveApi(config->name()))) {}
+
+Http::FilterHeadersStatus BridgingFilter::decodeHeaders(Http::RequestHeaderMap& headers,
+                                                        bool end_stream) {
+  // This is a hack that results in a full copy of the header map every time.
```

Comment:
Nit: this isn't really so much of a hack as an area for optimization, right? Can we file an issue and label it as tech debt?